inference/core/workflows/core_steps/models/foundation/google_vision_ocr/v1.py

```diff
@@ -0,0 +1,181 @@
+            raise ValueError(f"Invalid ocr_type: {ocr_type}")
+
+        response = requests.post(
+            "https://vision.googleapis.com/v1/images:annotate",
+            params={"key": api_key},
+            json={
+                "requests": [
+                    {
+                        "image": {"content": image.base64_image},
+                        "features": [{"type": type}],
+                    }
+                ]
+            },
+        )
+
+        if response.status_code != 200:
+            raise RuntimeError(
+                f"Request to Google Cloud Vision API failed: {str(response.json())}"
+            )
+
+        result = response.json()["responses"][0]
+
+        # Check for image without text
+        if "textAnnotations" not in result or not result["textAnnotations"]:
+            return {
+                "text": "",
+                "language": "",
+                "predictions": sv.Detections.empty(),
+            }
+
+        # Extract predictions from the response
+        text = result["textAnnotations"][0]["description"]
+        language = result["textAnnotations"][0]["locale"]
+
+        xyxy = []
+        confidence = []
+        classes = []
+        detections_id = []
+
+        for page in result["fullTextAnnotation"]["pages"]:
+            for block in page["blocks"]:
+                # Get bounding box coordinates
+                box = block["boundingBox"]["vertices"]
+                x_min = min(v["x"] for v in box)
+                y_min = min(v["y"] for v in box)
+                x_max = max(v["x"] for v in box)
+                y_max = max(v["y"] for v in box)
+                xyxy.append([x_min, y_min, x_max, y_max])
+
+                # Only DOCUMENT_TEXT_DETECTION provides confidence score, use 1.0 otherwise
+                confidence.append(block.get("confidence", 1.0))
+
+                # Get block text
+                block_text = []
+                for paragraph in block["paragraphs"]:
+                    for word in paragraph["words"]:
+                        word_text = "".join(
+                            symbol["text"] for symbol in word["symbols"]
+                        )
+                        block_text.append(word_text)
+                classes.append(" ".join(block_text))
+
+                # Create unique detection id for each block
+                detections_id.append(uuid4())
+
+        predictions = sv.Detections(
+            xyxy=np.array(xyxy),
+            confidence=np.array(confidence),
+            class_id=np.arange(len(classes)),
+            data={CLASS_NAME_DATA_FIELD: np.array(classes)},
+        )
+
+        predictions[DETECTION_ID_KEY] = np.array(detections_id)
+        predictions[PREDICTION_TYPE_KEY] = np.array(["ocr"] * len(predictions))
+        image_height, image_width = image.numpy_image.shape[:2]
+        predictions[IMAGE_DIMENSIONS_KEY] = np.array(
+            [[image_height, image_width]] * len(predictions)
+        )
+
```

Comment:
if you use [attach_parents_coordinates_to_sv_detections(...)](https://github.com/roboflow/inference/blob/6981108d866be40952fd32257da693cd8457e299/inference/core/workflows/core_steps/common/utils.py#L180) we would have full set of metadata in sv.detections

Reply:
Done 👍